### PR TITLE
Patch libh3 to include a document undefined check

### PR DESCRIPTION
**Pull request by 4nthonylin**
Address #117 and handles WebWorker as mentioned in #163. Note that this is a patch until #163 can be fully resolved.

### CODE REVIEW

**Review by dfellis — APPROVED** [review]

**Review by nrabinowitz — APPROVED** [review]

**Comment by nrabinowitz** [comment]
Thanks for submitting this - looks like a great workaround until we figure out how to upgrade emscripten.

File: test/test-build.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+test('validateLIBH3HasDocumentUndefinedFix', assert => {
+    // Validate that the bundle does not contain a bug that prevents h3-js to be imported into webworkers and react native.
+    // Problem was fixed in a later version of emscripten however there are significant performance regressions when upgrading.
+    // See #163 and #117 for more details.
+    const fileContents = fs.readFileSync(LIBH3_PATH, {encoding: 'utf8', flag: 'r'});
+    assert.doesNotMatch(fileContents, ORIGINAL_DOCUMENT_REGEX, 'bundle was not patched');
+    assert.match(fileContents, PATCHED_DOCUMENT_REGEX, 'bundle was not patched');
+    assert.end();
+});
```
Comment (nrabinowitz) [inline test/test-build.js:34]: I think the tests are failing on the lack of newline here

Comment (4nthonylin) [reply test/test-build.js]: Yep! Forgot to run prettier.

**Comment by 4nthonylin** [comment]
Just as a verification that webworker works:
```ts
// index.ts
const worker = new Worker(new URL('./worker.ts', import.meta.url), {type: "module"});
worker.onmessage = (e) => console.log(e);

worker.postMessage({type:"process_h3", value: [-120, 45]});
```

```ts
import {latLngToCell} from "h3-js";

self.onmessage = (e) => {
    const data = e.data;

    if (data.type == "process_h3") {
        const [lng, lat] = data.value;
        self.postMessage(latLngToCell(lat, lng, 5));
    }
}
```

```
// console
MessageEvent {isTrusted: true, data: '8528ab77fffffff', origin: '', lastEventId: '', source: null}
```
image

**Comment by nrabinowitz** [comment]
Thanks for contributing! I will try to push out a new release early this week.

**Pull request merged**